dotnet-desktop-guide/framework/winforms/controls/printpreviewdialog-control-overview-windows-forms.md

```diff
@@ -10,43 +10,96 @@ ms.assetid: efd4ee8d-6edd-47ec-88e4-4a4759bd2384
+Under the following conditions, the <xref:System.Windows.Forms.PrintPreviewDialog> control initializes slowly:
 
 - A network printer is used.
 - User preferences for this printer, such as duplex settings, are modified.
 
-For apps running on the .NET Framework 4.5.2, you can add the following key to the \<appSettings> section of your configuration file to improve the performance of <xref:System.Windows.Forms.PrintPreviewDialog> control initialization:
+The optimization isn't applied if you use the <xref:System.Drawing.Printing.PrintDocument.QueryPageSettings> event to modify page settings.
 
-```xml
-<appSettings>
-   <add key="EnablePrintPreviewOptimization" value="true" />
-</appSettings>
-```
+To apply the optimization, set the `Switch.System.Drawing.Printing.OptimizePrintPreview` runtime config option to `true`.
+
+# [.NET](#tab/dotnet)
+
+The option can be set in the runtimeconfig.json configuration file or the project file of an app:
+
+- Set in the _runtimeconfig.template.json_ source file.
+
+  To configure the default setting for your app, apply the setting in the _runtimeconfig.template.json_ source file. When the app is compiled or published, the template file is used to generate a runtime config file. Alternatively you can [apply the setting in the project file](#apply-the-setting-in-the-project-file).
+
+  ```json
+  {
+    "configProperties": {
+      "Switch.System.Drawing.Printing.OptimizePrintPreview": true
+    }
+  }
+  ```
+
+  For more information about runtime config, see [.NET runtime configuration settings](/dotnet/core/runtime-config/).
 
-If the `EnablePrintPreviewOptimization` key is set to any other value, or if the key is not present, the optimization is not applied. This key has no effect if the application is running on .NET Framework 4.6 or later.
+- Set in the _{appname}.runtimeconfig.json_ output file.
 
-For apps running on the .NET Framework 4.6 or later versions, you can add the following switch to the [\<AppContextSwitchOverrides>](/dotnet/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element) element in the [\<runtime>](/dotnet/framework/configure-apps/file-schema/runtime/index) section of your app config file:
+  To configure the published app, apply the setting in the _{appname}.runtimeconfig.json_ file's `runtimeOptions/configProperties` section.
+
+  ```json
+  {
+    "runtimeOptions": {
+      "configProperties": {
+        "Switch.System.Drawing.Printing.OptimizePrintPreview": true,
+      }
+    }
+  }
+  ```
+
+  For more information about runtime config, see [.NET runtime configuration settings](/dotnet/core/runtime-config/).
+
+- Set in the project file.
+
+  As an alternative to using the _runtimeconfig.template.json_ file, apply the setting in the project file. Add the `<RuntimeHostConfigurationOption>` setting to an `<ItemGroup>`:
+
+  ```xml
+  <Project Sdk="Microsoft.NET.Sdk">
+  
+    <!-- Other project settings ... -->  
+  
+    <ItemGroup>
```

Comment:
You also need a bool property that controls if the runtime config file is generated https://learn.microsoft.com/en-us/dotnet/core/runtime-config/#runtimeconfigjson

Reply:
Nevermind, GenerateRuntimeConfigurationFiles is for dlls only.